From: toddrme2178@gmail.com
Date: 2017-07-40
Subject: Fix implicit-pointer-decl warnings and implicit-fortify-decl errors

Fix the following warnings:

  Program is using implicit definitions of functions getting
  pointers or implemented by macros.
  W: python-scipy implicit-pointer-decl scipy/sparse/linalg/dsolve/SuperLU/SRC/dmach.c:52
  W: python-scipy implicit-pointer-decl scipy/sparse/linalg/dsolve/SuperLU/SRC/smach.c:52

And fix the following error:

  I: Program is using implicit definitions of special functions.
  E: python-scipy implicit-fortify-decl scipy/spatial/qhull_misc.h:23

---

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/dmach.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/dmach.c
@@ -1,6 +1,7 @@
 #include <float.h>
 #include <math.h>
 #include <stdio.h>
+#include <string.h>
 
 double dmach(char *cmach)
 {

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/smach.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/smach.c
@@ -1,6 +1,7 @@
 #include <float.h>
 #include <math.h>
 #include <stdio.h>
+#include <string.h>
 
 float smach(char *cmach)
 {

--- a/scipy/spatial/qhull_misc.h
+++ b/scipy/spatial/qhull_misc.h
@@ -2,6 +2,9 @@
  * Handle different Fortran conventions.
  */
 
+#include <stdio.h>
+#include <lapacke.h>
+
 #if defined(NO_APPEND_FORTRAN)
 #if defined(UPPERCASE_FORTRAN)
 #define F_FUNC(f,F) F
